packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -778,28 +778,20 @@ export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
         this.baseGCDetailsP = new LazyPromise<IGarbageCollectionDetailsBase>(async () => {
             return (await props.getBaseGCDetails()) ?? {};
         });
+
+        if (props.snapshotTree !== undefined) {
+            this.summarizerNode.updateBaseSummaryState(props.snapshotTree);
```

Comment:
> this.summarizerNode.updateBaseSummaryState(props.snapshotTree); [](http://example.com/codeflow?start=11&length=64)

The change looks good - I could not find any call site with snapshotValue as string as well. 

I'm curious - before, we would call the loadBaseSummary (now updateBaseSummaryState) during the DS realization (realizationCore). Is it possible the summarizerNode hierarchy might change in between the constructor and the realization? Like new child nodes added ? If so, we could have, by the time the realization happens, some nodes for which we did not have this._latestSummary set and would not set additionalPath for.

Reply:
It's not possible to create a child summarizer node without realizing the data store first.
Even if it did, this code would work just fine. When creating a child summarizer node, the parent summarizer node passes on its latest summary to the child (see [here](https://github.com/microsoft/FluidFramework/blob/6a8d3c085897b73806cd5f4c8c3179b4887f8617/packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts#L482) and [here](https://github.com/microsoft/FluidFramework/blob/6a8d3c085897b73806cd5f4c8c3179b4887f8617/packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts#L535)).

The general idea is that once a summarizer node's latest summary is updated, any child created after this time will inherit the latest summary. The bug that this PR is trying to fix is that the latest summary of the data store node was updated later than it should.

Reply:
> Also, the tree would be different as well, wouldn't it ?
Not sure what you mean by this. Can you please elaborate?

Reply:
> Not sure what you mean by this. Can you please elaborate?

You answered in the previous post. We should be good.